### ISSUE TITLE
Unhandled exception: Invalid Int64

### ISSUE THREAD

**Issue by syalon** [crystal-lang/crystal#11490]
The following code parse error, expect to get the correct serialized value.

```
require "json"
p! JSON.parse({a: 10649017208589841323_u64}.to_json)
```

```
Unhandled exception: Invalid Int64: 10649017208589841323 at line 1, column 6 (JSON::ParseException)
  from /usr/lib/crystal/json/token.cr:25:7 in 'int_value'
  from /usr/lib/crystal/json/parser.cr:21:28 in 'parse_value'
  from /usr/lib/crystal/json/parser.cr:85:19 in 'parse_object'
  from /usr/lib/crystal/json/parser.cr:35:7 in 'parse_value'
  from /usr/lib/crystal/json/parser.cr:13:12 in 'parse'
  from /usr/lib/crystal/json.cr:135:5 in 'parse'
  from /eval:3:4 in '__crystal_main'
  from /usr/lib/crystal/crystal/main.cr:110:5 in 'main_user_code'
  from /usr/lib/crystal/crystal/main.cr:96:7 in 'main'
  from /usr/lib/crystal/crystal/main.cr:119:3 in 'main'
  from __libc_start_main
  from _start
  from ???
```

**Comment by Blacksmoke16** [comment]
Duplicate of https://github.com/crystal-lang/crystal/issues/8830

**Comment by Blacksmoke16** [comment]
Going to reopen as this still seems to be an issue even after #8830 fixed it for `JSON::PullParser`. `JSON.parse` should also be able to handle these larger numeric types.

**Comment by asterite** [comment]
Sounds good, but I don't know how we can do it. Maybe we can introduce another type to `JSON::Any`, which is `LargeNumber, value: String`.

**Comment by straight-shoota** [comment]
Closing this in favour of #12953
